server/analyzer/replace_node.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analyzer
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/analyzer"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/dolthub/go-mysql-server/sql/transform"
+
+	pgnodes "github.com/dolthub/doltgresql/server/node"
+)
+
+// ReplaceNode is used to replace generic top-level nodes with Doltgres versions that wrap them, without performing any
+// additional analysis. This is used to handle relatively straightforward tasks, like delete cascading, etc.
+func ReplaceNode(ctx *sql.Context, a *analyzer.Analyzer, node sql.Node, scope *plan.Scope, selector analyzer.RuleSelector, qFlags *sql.QueryFlags) (sql.Node, transform.TreeIdentity, error) {
```

Comment:
We probably don't want this.

Doltgres needs its own DatabaseProvider / Database implementations that handle these operations, so it can insert its own logic. Doing it by monkey patching the query plan has all sorts of issues.

Reply:
I don't quite see how that would line up. `DatabaseProvider` and `Database` have no bearing on `CreateSequence`, `CreateType`, etc. These are specific to the nodes more so than something general on the database.